client/state/user-settings/reducer.js

```diff
@@ -42,7 +44,19 @@ export const unsavedSettings = ( state = {}, action ) => {
 	}
 };
 
+export const pendingPasswordChange = ( state = false, action ) => {
+	switch ( action.type ) {
+		case USER_SETTINGS_SAVE:
+		case USER_SETTINGS_UPDATE:
+		case USER_SETTINGS_UPDATE_FAILURE: {
+			return action.type === USER_SETTINGS_SAVE && Boolean( action.settingsOverride?.password );
+		}
+	}
+	return state;
+};
+
 export default combineReducers( {
 	settings,
 	unsavedSettings,
+	pendingPasswordChange,
```

Comment:
Naming nit: this seems like it's pre-empting the selector that is going to use it. Should we name it like `updatingPassword` or something? And then the selector would be `isPendingPasswordChange` as it is.

Reply:
addressed in 54d94165334069c4f859c751b770866a0dba12f2